packages/json-rpc-middleware-stream/src/createStreamMiddleware.ts

```diff
@@ -147,8 +147,10 @@ export default function createStreamMiddleware(options: Options = {}) {
           `StreamMiddleware - Retry limit exceeded for request id "${req.id}"`,
         );
       }
-
-      idMap[req.id].retryCount = retryCount + 1;
+      const idMapObject = idMap[req.id];
+      if (idMapObject) {
+        idMapObject.retryCount = retryCount + 1;
+      }
```

Comment:
Nit: To make this diff smaller, can you also say this as this?

``` suggestion
      idMap[req.id]?.retryCount = retryCount + 1;

Reply:
I think optional chaining can't be used on the LHS of an assignment.

Reply:
Thank you 🤦🏻 

![](https://media4.giphy.com/media/10o3Um2U3wa4DK/giphy.gif?cid=ecf05e4785zw7lmjodm83vwtm64cztp64sninyge16u9c3xq&ep=v1_gifs_search&rid=giphy.gif&ct=g)